pkg/apis/openebs.io/scp/v1alpha1/storage-pool.go

```diff
@@ -29,8 +31,387 @@ import (
+	State StoragePoolState `json:"state,omitempty"`
+}
+
+// StoragePoolType of the storage pool, viz lvm, spdk blobstor
+type StoragePoolType string
+
+const (
+	// LVM2StoragePool for lvm pools
+	LVM2StoragePool StoragePoolType = "lvm2"
+
+	// SpdkBlobstorStoragePool for spdk-blobstor pools
+	SpdkBlobstorStoragePool StoragePoolType = "spdk-blobstor"
+)
+
+// StoragePoolCofiguration for the defined type of the pool
+type StoragePoolCofiguration struct {
+	// LVM pool related configuration
+	// +nullable
+	LVM *LVMPoolConfiguration
+
+	// SpdkBlobStor pool related configuration
+	// +nullable
+	SpdkBlobStor *SpdkBlobStorPoolConfiguration
+}
+
+// LVMPoolConfiguration for LVM pool related configuration
+type LVMPoolConfiguration struct {
+	//TODO: Decide the sub fields
+}
+
+// SpdkBlobStorPoolConfiguration for SpdkBlobStor pool related configuration
+type SpdkBlobStorPoolConfiguration struct {
+	//TODO: Decide the sub fields
+}
+
+// StoragePoolDeviceSpec for the configuration and filtering related specs for devices
+type StoragePoolDeviceSpec struct {
+	// MaxDeviceCount for maximum allowed devices
+	MaxDeviceCount uint64 `json:"maxDeviceCount,omitempty"`
+
+	// DeviceTypeIdentifier for the device viz.  ssd-4k, ssd-16k, nvme-64k
+	DeviceTypeIdentifier StoragePoolDeviceTypeIdentifier `json:"deviceTypeIdentifier,omitempty"`
+
+	// DeviceSelector for labels to match with devices incase they are labeled
+	DeviceSelector map[string]string `json:"deviceSelector,omitempty"`
+
+	// DeviceSelector for filtering devices based on FiltersOptions
+	DeviceFilters map[string][]FilterOptions `json:"deviceFilters,omitempty"`
+
+	// Devices for the UIDs of the free devices
+	Devices []types.UID `json:"devices,omitempty"`
+}
+
+type StoragePoolCapabilities struct {
+	// DataStorageCapabilities for the storage pool
+	DataStorageCapabilities StoragePoolDataStorageCapabilities `json:"dataStorageCapabilities,omitempty"`
+
+	// DataSecurityCapabilities for the storage pool
+	DataSecurityCapabilities StoragePoolDataSecurityCapabilities `json:"dataSecurityCapabilities,omitempty"`
+
+	// IOConnectivityCapabilities for the storage pool
+	IOConnectivityCapabilities StoragePoolIOConnectivityCapabilities `json:"ioConnectivityCapabilities,omitempty"`
+
+	// IOPerformanceCapabilities for the storage pool
+	IOPerformanceCapabilities StoragePoolIOPerformanceCapabilities `json:"ioPerformanceCapabilities,omitempty"`
+
+	// DataProtectionCapabilities for the storage pool
+	DataProtectionCapabilities StoragePoolDataProtectionCapabilities `json:"dataProtectionCapabilities,omitempty"`
+}
+
+// StoragePoolReferenceResource
+type StoragePoolReferenceResource struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+}
+
+// StoragePoolStorageCapacity for the pool,viz total, used and available capacity
+type StoragePoolStorageCapacity struct {
+	// TotalCapacity of the pool
+	TotalCapacity resource.Quantity `json:"totalCapacity,omitempty"`
```

Comment:
keep the name and  json tag as toal Capacity is repetitive here

Reply:
same for used and available

Reply:
Sure. Removing Capacity from Variable names.